internal/querycoord/channel_allocator.go

```diff
@@ -31,13 +33,17 @@ func defaultChannelAllocatePolicy() ChannelAllocatePolicy {
 }
 
 // ChannelAllocatePolicy helper function definition to allocate dmChannel to queryNode
-type ChannelAllocatePolicy func(ctx context.Context, reqs []*querypb.WatchDmChannelsRequest, cluster Cluster, wait bool, excludeNodeIDs []int64) error
+type ChannelAllocatePolicy func(ctx context.Context, reqs []*querypb.WatchDmChannelsRequest, cluster Cluster, metaCache Meta, wait bool, excludeNodeIDs []int64) error
 
-func shuffleChannelsToQueryNode(ctx context.Context, reqs []*querypb.WatchDmChannelsRequest, cluster Cluster, wait bool, excludeNodeIDs []int64) error {
+func shuffleChannelsToQueryNode(ctx context.Context, reqs []*querypb.WatchDmChannelsRequest, cluster Cluster, metaCache Meta, wait bool, excludeNodeIDs []int64) error {
+	if len(reqs) == 0 {
+		return nil
+	}
 	for {
-		availableNodes, err := cluster.onlineNodes()
-		if err != nil {
-			log.Debug(err.Error())
+		availableNodes := cluster.onlineNodes()
```

Comment:
There is a inconsitency here. 
clusert.OfflineNodes return error if no node available.
cluster.onlineNodes return only empty list and the err is handled outside, which mode do you prefer?

Reply:
I prefer the latter, and it would be more flexible to handle the error outside.